### PR TITLE
Add members and metadata to the generated groups of each array.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
First part of SC-26763.

### CODE REVIEW

File: src/test_array_builder.cc [hunk 1]
```diff
@@ -944,6 +945,32 @@ bool build_heterogeneous_arrays(
...
+  g.put_metadata(key + "_multi", datatype, values.size(), values.data());
+};
+
+void build_group(Context &ctx, const std::string &group_base) {
+  Group g{ctx, group_base, tiledb_query_type_t::TILEDB_WRITE};
+
+  put_metadata<uint8_t>(g, "u8", TILEDB_UINT8, 77);
+  put_metadata<uint16_t>(g, "u16", TILEDB_UINT16, 777);
+  put_metadata<uint32_t>(g, "u32", TILEDB_UINT32, 77777);
+  put_metadata<uint64_t>(g, "u64", TILEDB_UINT64, 7777777777);
```
Comment (ihnorton) [inline src/test_array_builder.cc]: Let's add a string and bool too, at least

**Comment by Shelnutt2** [comment]
Is there any opportunity to have nested groups so we can test the member type == group?

**Comment by teo-tsirpanis** [comment]
@Shelnutt2 done.

File: src/test_array_builder.cc [hunk 2]
```diff
@@ -944,6 +945,42 @@ bool build_heterogeneous_arrays(
...
+}
+
+template <class T>
+void put_metadata(Group &g, const std::string &key,
+                  tiledb_datatype_t datatype, T value) {
+  g.put_metadata(key, datatype, 1, &value);
+
+  std::array<T, 7> values;
+  std::fill(values.begin(), values.end(), value);
+  g.put_metadata(key + "_multi", datatype, static_cast<uint32_t>(values.size()), values.data());
```
Comment (dhoke4tdb) [inline src/test_array_builder.cc]: should 3rd param 'values.size()' be 'values.size()*sizeof(T)', 
or actually maybe should be 'sizeof(values)'?
(I.e., is param supposed to be count of items, or number of bytes?)

File: src/test_array_builder.cc [hunk 3]
```diff
@@ -944,6 +945,42 @@ bool build_heterogeneous_arrays(
...
+
+void build_group(Context &ctx, const std::string &group_base) {
+  Group g{ctx, group_base, tiledb_query_type_t::TILEDB_WRITE};
+
+  put_metadata<bool>(g, "bool", TILEDB_BOOL, true);
+  put_metadata<uint8_t>(g, "u8", TILEDB_UINT8, 77);
+  put_metadata<uint16_t>(g, "u16", TILEDB_UINT16, 777);
+  put_metadata<uint32_t>(g, "u32", TILEDB_UINT32, 77777);
+  put_metadata<uint64_t>(g, "u64", TILEDB_UINT64, 7777777777);
+  put_metadata(g, "str", "77777");
```
Comment (dhoke4tdb) [inline src/test_array_builder.cc:971]: Consider making this some number of '7's different from number of digits in any of above numeric values...

**Review by dhoke4tdb — CHANGES_REQUESTED** [review]
Please confirm noted values.size() is correct as coded, or change if needed.

File: src/test_array_builder.cc (hunk 2 above)
Comment (teo-tsirpanis) [reply src/test_array_builder.cc]: And I was wondering why I couldn't see the sevens in the hex editor. 🤦🏻‍♂️ Thanks!

(we need a generic API for this)

**Comment by Shelnutt2** [comment]
> @Shelnutt2 done.

Thanks!

**Review by ihnorton — APPROVED** [review]
👍

File: src/test_array_builder.cc (hunk 2 above)
Comment (teo-tsirpanis) [reply src/test_array_builder.cc]: Done.

File: src/test_array_builder.cc (hunk 3 above)
Comment (teo-tsirpanis) [reply src/test_array_builder.cc:971]: Done, I used hex values for the integers.

**Comment by teo-tsirpanis** [comment]
How to add the generated arrays to the PR? ~~For recent Core versions it looks like not all the combinations are present.~~

**Comment by ihnorton** [comment]
Run the generator, the content will be created in the current working directory; then move the (new) content into the vX_Y_Z folder. Make sure you run the generator with core built against the correct branch, in this case 2.15.0 or 2.15.1 (which will be released later today).

> For recent Core versions it looks like not all the combinations are present.

Which ones?

**Comment by teo-tsirpanis** [comment]
> Which ones?

Never mind, I got misled by the file count; the directories for versions prior to 2.12 had much more files, but all recent versions have 132 arrays.

**Comment by teo-tsirpanis** [comment]
Here's what I'm talking about. The test arrays I just generated are much bigger than before.

image

We have 816 attributes? 🤔

image

---

But yeah I'll merge.

**Pull request merged**